src/ORM/DataQuery.php

```diff
@@ -447,6 +448,34 @@ public function count()
         return $this->getFinalisedQuery()->count("DISTINCT {$quotedColumn}");
     }
 
+    /**
+     * Return whether this dataquery will have records. This will use `EXISTS` statements in SQL which are more
+     * performant - especially when used in combination with indexed columns (that you're filtering on)
+     *
+     * @return bool
+     */
+    public function exists()
+    {
+        // Grab a statement selecting "everything" - the engine shouldn't care what's being selected in an "EXISTS"
+        // statement anyway
+        $statement = $this->getFinalisedQuery(null, true);
+        $statement->setSelect('*');
+
+        // Clear distinct, grouping, order, and having as it's not relevant for an exists query
+        $statement->setDistinct(false);
+        $statement->setOrderBy(null);
+        $statement->setGroupBy(null);
+        $statement->setHaving(null);
```

Comment:
Having and group by absolutely will be necessary, as it will affect whether exists() returns rows or not for matching groups.

Reply:
And with grouping, you need to respect the original select columns don't you? :P

Reply:
E.g. `SELECT "Value", COUNT("Value") as "CountValue" FROM "Table" GROUP BY "Value" HAVING "CountValue" > 1` should return false for exists if no item has more than one value, but true as soon as a value is in the table twice.

Reply:
With or without the group there'll be at least one record right? Having is necessary though. I think that's a mistake.

Reply:
Right, if we keep the having then the group is relevant

Reply:
I don't think the original columns ever matter though - records are not filtered out based on what you select - I think. You either have records or you don't. Respecting `count(*), ParentID` or something when `GROUP BY ParentID` doesn't matter because you either have rows or you don't.

Reply:
> Right, if we keep the having then the group is relevant

I'm confused - the having doesn't rely on the group. So I'm back to the group isn't required 😆

Reply:
I've removed the `having`. I did a little research to back up my thoughts that `GROUP BY` remains superfluous in these `EXISTS` queries - and I'm pretty sure it is. I can't think of any scenario where adding a `GROUP BY` will change the returned records from "some" to "none".

Reply:
How does your having work if you don't group anymore then?

Reply:
Aggregates in `HAVING` don't require a `GROUP BY`?

Reply:
I googled this to be double sure I wasn't making a bad assumption - here's the stack exchange result I got 😂 :

https://dba.stackexchange.com/questions/57445/use-of-having-without-group-by-in-sql-queries

Reply:
Ok this _can_ break if you're doing your own aggregates an aliasing I guess